Utilities/DataLoading/EdFi.LoadTools.Test/SmokeTests/APIGetTests.cs

```diff
@@ -75,78 +81,99 @@ public class ApiGetTests
+                                        {
+                                            var skip = int.Parse(context.Request.Query["offset"]);
+                                            var take = int.Parse(context.Request.Query["limit"]);
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(_data.Skip(skip).Take(take)));
+                                        }
+                                        else if (!string.IsNullOrEmpty(context.Request.Query[KeyName]))
+                                        {
+                                            var keyValue = context.Request.Query[KeyName];
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(
+                                                    _data.SingleOrDefault(
+                                                        x =>
+                                                            x[KeyName].Value<string>() == keyValue)));
+                                        }
+                                        else if (!string.IsNullOrEmpty(context.Request.Query[PropertyName]))
+                                        {
+                                            var propertyValue = context.Request.Query[PropertyName];
+
+                                            await context.Response.WriteAsync(
+                                                JsonConvert.SerializeObject(
+                                                    _data.Where(
+                                                        x =>
+                                                            x[PropertyName].Value<string>() == propertyValue)));
+                                        }
+                                        else
+                                        {
+                                            await context.Response.WriteAsync(JsonConvert.SerializeObject(_data));
+                                        }
+                                    }));
+                        })
+                ;
+
+            Host = hostBuilder.Build();
+            Host.StartAsync();
         }
 
         [OneTimeTearDown]
         public void Cleanup()
         {
-            _webApp.Dispose();
+            Host.StopAsync();
+            Host.Dispose();
         }
 
         [Test]
         public void GetAll_should_store_results_in_dictionary()
         {
             var dictionary = new Dictionary<string, JArray>();
 
-            var subject = new GetAllTest(_resource, dictionary, _configuration, tokenHandler);
+            var client = Host.GetTestClient();
+            client.BaseAddress = new System.Uri(Address);
+
+            var subject = new GetAllTest(_resource, dictionary, _configuration, tokenHandler, client);
             var result = subject.PerformTest().Result;
```

Comment:
should this be await subject.PerformTest() ?

Reply:
Updated